src/Stack/Docker.hs

```diff
@@ -243,8 +245,17 @@ runContainerAndExit getCmdArgs
      (env,isStdinTerminal,isStderrTerminal,homeDir) <- liftIO $
        (,,,)
        <$> getEnvironment
+-- hIsTerminaDevice does not recognise input or output handles to mintty
+-- terminals as terminal devices, but isMinTTYHandle does.
+#ifdef WINDOWS
+       <*> ((||) <$> hIsTerminalDevice stdin
```

Comment:
Can I recommend that instead of including conditionals in multiple places, you define a helper function once which uses CPP, and use that in all of these places?

Additionally, it can be _marginally_ more efficient to write the code as something like:

```haskell
x <- hIsTerminalDevice stdin
if x then return True else withHandleToHANDLE stdin isMinTTYHandle
```

This will only run the second action if the first returns `False`.

Reply:
I've used a helper function `hIsTerminalDeviceOrMinTTY`, which I added to module `Stack.Prelude` - I could not see a better place.